src/MotorPlugin.cc

```diff
@@ -0,0 +1,704 @@
+/*
+   Copyright (C) 2025 ArduPilot.org
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "MotorPlugin.hh"
+
+#include <gz/msgs/entity_factory.pb.h>
+
+#include <memory>
```

Comment:
There are some missing includes here for `std::` namespace classes that are used. A linter should pick this up.